sdk/python/tests/expediagroup/test_milvus_online_store.py

```diff
@@ -1,37 +1,293 @@
+        ]
+        mock_logger.info.assert_has_calls(expected_log_calls)
+
+        with pytest.raises(Exception):
+            with MilvusConnectionManager(repo_config.online_store):
+                raise Exception("Test Exception")
+        mock_logger.error.assert_called_once()
+
+
+class TestMilvusOnlineStore:
+
+    collection_to_write = "Collection2"
+    collection_to_delete = "Collection1"
+
+    def setup_method(self, milvus_online_setup):
+        # Ensuring that the collections created are dropped before the tests are run
+        connections.connect(alias="default", host=HOST, port=19530)
+        # Dropping collections if they exist
+        if utility.has_collection(self.collection_to_delete):
+            utility.drop_collection(self.collection_to_delete)
+        if utility.has_collection(self.collection_to_write):
+            utility.drop_collection(self.collection_to_write)
+        # Closing the temporary collection to do this
+        connections.disconnect("default")
+
+    def test_milvus_update_add_collection(
+        self, repo_config, milvus_online_setup, caplog
+    ):
+        # Creating a common schema for collection
+        schema = CollectionSchema(
+            fields=[
+                FieldSchema(
+                    "int64", DataType.INT64, description="int64", is_primary=True
+                ),
+                FieldSchema(
+                    "float_vector", DataType.FLOAT_VECTOR, is_primary=False, dim=128
+                ),
+            ]
+        )
+
+        MilvusOnlineStore().update(
+            config=repo_config,
+            tables_to_delete=[],
+            tables_to_keep=[
+                MockFeatureView(name=self.collection_to_write, schema=schema)
             ],
-            "auto_id": True,
-        }
+            entities_to_delete=None,
+            entities_to_keep=None,
+            partial=None,
+        )
+        assert len(utility.list_collections()) == 1
+        assert utility.has_collection(self.collection_to_write) is True
+        assert (
+            f"Collection {self.collection_to_write} has been created successfully."
+            in caplog.text
+        )
 
-        milvus.create_collection(collection_name, fields)
+    def test_milvus_update_add_existing_collection(
+        self, repo_config, caplog, milvus_online_setup
+    ):
 
-    collection = milvus.describe_collection(collection_name)
-    assert collection.get("collection_name") == collection_name
+        # Creating a common schema for collection
+        schema = CollectionSchema(
+            fields=[
+                FieldSchema(
+                    "int64", DataType.INT64, description="int64", is_primary=True
+                ),
+                FieldSchema(
+                    "float_vector", DataType.FLOAT_VECTOR, is_primary=False, dim=128
+                ),
+            ]
+        )
 
-    online_store_creator.teardown()
+        MilvusOnlineStore().update(
```

Comment:
using the milvus online store here makes this test dependent on `update()` working as expected. Instead creating a collection via the pymilvus sdk directly would be better here to simulate that a collection already exists.